Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,406 @@
+  });
+  connect(m_reset_btn, &QPushButton::pressed, this, &CodeDiffDialog::ClearData);
+  connect(m_matching_results_list, &CodeDiffDialog::customContextMenuRequested, this,
+          &CodeDiffDialog::OnContextMenu);
+}
+
+void CodeDiffDialog::ClearData()
+{
+  if (m_record_btn->isChecked())
+    m_record_btn->toggle();
+  ClearBlockCache();
+  m_matching_results_list->clear();
+  m_exclude_size_label->setText(QStringLiteral("Excluded: 0"));
+  m_include_size_label->setText(QStringLiteral("Included: 0"));
+  m_exclude_btn->setEnabled(false);
+  m_include_btn->setEnabled(false);
+  // Swap is used instead of clear for efficiency in the case of huge m_include/m_exclude
+  std::vector<Diff>().swap(m_include);
+  std::vector<Diff>().swap(m_exclude);
+  JitInterface::SetProfilingState(JitInterface::ProfilingState::Disabled);
+}
+
+void CodeDiffDialog::ClearBlockCache()
+{
+  Core::State old_state = Core::GetState();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Paused);
+
+  JitInterface::ClearCache();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Running);
+}
+
+void CodeDiffDialog::OnRecord(bool enabled)
+{
+  JitInterface::ProfilingState state;
+
+  if (enabled)
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Recording..."));
+    state = JitInterface::ProfilingState::Enabled;
+    m_exclude_btn->setEnabled(true);
+    m_include_btn->setEnabled(true);
+  }
+  else
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Start Recording"));
+    state = JitInterface::ProfilingState::Disabled;
+    m_exclude_btn->setEnabled(false);
+    m_include_btn->setEnabled(false);
+  }
+
+  m_record_btn->update();
+  JitInterface::SetProfilingState(state);
+}
+
+void CodeDiffDialog::OnInclude()
+{
+  const auto recorded_symbols = CalculateSymbolsFromProfile();
+  if (m_include.empty() && m_exclude.empty())
+  {
+    m_include = recorded_symbols;
+    return;
+  }
+
+  std::vector<Diff> current_diff;
+  if (m_exclude.empty())
+  {
+    current_diff = recorded_symbols;
+  }
+  else
+  {
+    for (auto& iter : recorded_symbols)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+      }
+    }
+  }
+
+  if (!m_include.empty())
+  {
+    m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                   [&](Diff& v) {
+                                     // Remove values if it can't find matching symbols
+                                     return std::none_of(
+                                         current_diff.begin(), current_diff.end(),
+                                         [&](Diff& p) { return p.symbol == v.symbol; });
+                                   }),
+                    m_include.end());
+  }
+  else
+  {
+    m_include = recorded_symbols;

```

Comment:
This does not account for staying active in the recorded state while executing this filter. The profiler still could have new symbols since entering the last function. If we use the above filter we may lose functions in the last few frames.